providers/jackson2/src/main/java/org/jboss/resteasy/plugins/providers/jackson/ResteasyJackson2Provider.java

```diff
@@ -72,16 +72,15 @@ public boolean isWriteable(Class<?> aClass, Type type, Annotation[] annotations,
 
    private static class ClassAnnotationKey
    {
-      private AnnotationArrayKey annotations;
+      private  Annotation[] annotations;
       private ClassKey classKey;
       private int hash;
 
       private ClassAnnotationKey(final Class<?> clazz, final Annotation[] annotations)
       {
-         this.annotations = new AnnotationArrayKey(annotations);
          this.classKey = new ClassKey(clazz);
-         hash = this.annotations.hashCode();
-         hash = 31 * hash + classKey.hashCode();
+         this.annotations = annotations;
+         hash = Arrays.hashCode(this.annotations) + classKey.hashCode();
```

Comment:
> Hi @jimma , besides the comment below, I just want to make sure that you thought carefully about the implications of changing / removing the AnnotationArrayKey concept. Please check this [6866e3f](https://github.com/resteasy/Resteasy/commit/6866e3ff2cd74402fb7250c50690ab31c39271a2) if you didn't already and ensure the changes you're proposing here won't restore the problem solved in that commit nor what led Jackson having the AnnotationBundleKey abstraction. If you already checked this all and believe we're safe ... great ;-)

I believe we don't have this problem because we completely remove the AnnotationArrayKey, but we still keep the code to 
check annotation equals with:
```
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ClassAnnotationKey that = (ClassAnnotationKey) o;

        if (!annotations.equals(that.annotations)) return false;
        if (!classKey.equals(that.classKey)) return false;

        return true;
    }
```

Reply:
> @jimma are you removing the multiply for 31 on purpose (if yes, why) ?

 Arrays.hashCode(this.annotations) is better than multiply with 31. Arrays.hashCode(annotations) simply does this:
```
    public static int hashCode(Object a[]) {
        if (a == null)
            return 0;

        int result = 1;

        for (Object element : a)
            result = 31 * result + (element == null ? 0 : element.hashCode());

        return result;
    }
```